ext/nnstreamer/tensor_filter/tensor_filter_executorch.cc

```diff
@@ -177,61 +150,69 @@ executorch_subplugin::configure_instance (const GstTensorFilterProperties *prop)
-      ET_LOG (Error, "Failed to parse model file %s", model_path);
-      return;
-    }
-    ET_LOG (Info, "Model file %s is loaded.", model_path);
-
-    // Use the first method in the program.
-    const auto method_name_result = program->get ().get_method_name (0);
-    ET_CHECK_MSG (method_name_result.ok (), "Program has no methods");
-    method_name = *method_name_result;
-    ET_LOG (Info, "Using method %s", method_name);
-
-    // MethodMeta describes the memory requirements of the method.
-    Result<MethodMeta> method_meta = program->get ().method_meta (method_name);
-    ET_CHECK_MSG (method_meta.ok (), "Failed to get method_meta for %s: 0x%" PRIx32,
-        method_name, (uint32_t) method_meta.error ());
-
-    method_allocator
-        = new MemoryAllocator (sizeof (method_allocator_pool), method_allocator_pool);
-
-
-    size_t num_memory_planned_buffers = method_meta->num_memory_planned_buffers ();
-    for (size_t id = 0; id < num_memory_planned_buffers; ++id) {
-      // .get() will always succeed because id < num_memory_planned_buffers.
-      size_t buffer_size = static_cast<size_t> (
-          method_meta->memory_planned_buffer_size (id).get ());
-      ET_LOG (Info, "Setting up planned buffer %zu, size %zu.", id, buffer_size);
-      planned_buffers.push_back (std::make_unique<uint8_t[]> (buffer_size));
-      planned_spans.push_back ({ planned_buffers.back ().get (), buffer_size });
+    module = std::make_unique<Module> (model_path, Module::MlockConfig::NoMlock);
+    if (module->load () != Error::Ok) {
+      const std::string err_msg
+          = "Failed to load module with Given file " + (std::string) model_path;
+      throw std::invalid_argument (err_msg);
     }
-    planned_memory = new HierarchicalAllocator (
-        { planned_spans.data (), planned_spans.size () });
-
-    // Assemble all of the allocators into the MemoryManager that the Executor
-    // will use.
-    memory_manager = new MemoryManager (method_allocator, planned_memory);
 
-    //
-    // Load the method from the program, using the provided allocators. Running
-    // the method can mutate the memory-planned buffers, so the method should
-    // only be used by a single thread at at time, but it can be reused.
-    //
+    ET_CHECK_MSG (module->is_loaded (), "Making module failed");
+
+    const auto forward_method_meta = module->method_meta ("forward");
+    ET_CHECK_MSG (forward_method_meta.ok (), "Getting method meta failed");
+
+    /** @todo support more data types */
+    auto convertType = [] (ScalarType type) {
+      switch (type) {
+        case ScalarType::Float:
+          return _NNS_FLOAT32;
+        default:
+          return _NNS_END;
+      }
+    };
+
+    // parse input tensors info
+    size_t num_inputs = forward_method_meta->num_inputs ();
+    inputInfo.num_tensors = num_inputs;
+    for (size_t i = 0; i < num_inputs; ++i) {
+      const auto input_meta = forward_method_meta->input_tensor_meta (i);
+
```

Comment:
let's use gst_tensors_info_get_nth_info() to get nth tensor-info ptr, to prevent index error.

Reply:
Oops I missed that, I updated the code PTAL